### PR TITLE
Versioneer

### LINKED ISSUE #1300: git version label yields version in violation of PEP 440
When an `xarray` installation does not match a released version ,it has a version number like `0.9.1-28-g769f120`.

This violates [PEP 440](https://www.python.org/dev/peps/pep-0440/), which leads to multiple problems:

* `pip install --update` will revert `xarray` back to `0.9.1`, because it does not recognise that `0.9.1-28-g769f120 > 0.9.1`
* packages with an `xarray` dependencies will be considered not satisfied.  Running a script through `setuptools` `load_entry_point` fails with `pkg_resources.ContextualVersionConflict: (xarray 0.9.1-28-g769f120 (/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages), Requirement.parse('xarray>=0.8')`

Instead, the version number above should be written as `0.9.1+r10345` or so, which would satisfy PEP 440 and not cause problems with `pip` or `setuptools`.

### PR DESCRIPTION
**Pull request by jhamman**
- [x] Closes #1300 (in a more portable way)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

This eliminates the need to edit `setup.py` before / after release and is a nice step towards simplifying xarray's release process.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -102,7 +102,7 @@
 # built documents.
 #
 # The short X.Y version.
-version = xarray.version.short_version
+version = xarray.__version__.split('+')[0]
```
Comment (jhamman) [inline doc/conf.py]: this is the one change that worries me. It is possible people were relying on the `xarray.version` module...

**Comment by dopplershift** [comment]
Versioneer has worked great for us. Cutting a release is triggered just by making a new release on GitHub.

**Comment by shoyer** [comment]
Very nice!

I tested this out locally and it seems to properly with picking up new tags 👍

File: doc/conf.py (hunk 1 above)
Comment (shoyer) [reply doc/conf.py]: If they were... then they will have to fix up their code.

**Pull request merged**